gradle/scripts/lib/java.gradle

```diff
@@ -48,45 +59,12 @@ configure(projectsWithFlags('java')) {
                     def useIncrementalCompilation = deps.size() == 0
                     options.incremental = useIncrementalCompilation
                     logger.info("${useIncrementalCompilation ? 'Enabling' : 'Disabling'} " +
-                                "incremental compilation for '${task.path}'")
+                            "incremental compilation for '${task.path}'")
                 }
             }
         }
     }
 
-    // Generate a source JAR.
-    task sourceJar(type: Jar) {
-        classifier = 'sources'
-        from sourceSets.main.allSource
-    }
-
-    // Generate a javadoc JAR.
-    task javadocJar(type: Jar, dependsOn: tasks.javadoc) {
-        classifier = 'javadoc'
-        from javadoc.destinationDir
-    }
-
-    // Generate source/javadoc JARs only when publishing.
```

Comment:
Do we now always generate source/javadoc JARs?

Reply:
It's always generated for `assemble` (and so `build`). I find this more idiomatic and not a huge problem - it only affects a curent target so e.g., `:grpc:assemble` only builds javadoc for `grpc`, not `core`. And IntelliJ Gradle Runner doesn't build. I was a bit surprised that our CI wasn't building javadoc, and I think it's good to verify javadoc builds in CI too.

But not hard to restore the skipping behavior if you still prefer that let me know.

Reply:
OK as it is.